app/controllers/api/v1/merchants/revenue_controller.rb

```diff
@@ -0,0 +1,12 @@
+class Api::V1::Merchants::RevenueController < ApplicationController
+
+  def show
+    if params[:date]
+      render json: Merchant.find(params[:id]).date_total_revenue(params[:date]),
+      serializer: SingleMerchantSerializer
+    else
+     render json: Merchant.find(params[:id]).total_revenue,
```

Comment:
rather than having to find the `Merchant` object twice, could we find it once, call the appropriate method, then render the json after?